cwatch.php

```diff
@@ -189,116 +189,67 @@ public function addService(
-                    // Get a count of licenses to provide for each type
-                    foreach ($license_types as $license_type => $value) {
-                        $license_types[$license_type] = isset($vars['configoptions'][$license_type])
-                            ? $vars['configoptions'][$license_type]
-                            : 0;
-                    }
-
-                    // Determine what term to add these licenses for
-                    $license_term = 'MONTH_1';
-                    if (($pricing = $this->Services->getPackagePricing($vars['pricing_id']))) {
-                        $license_term = strtoupper($pricing->period) . '_' . $pricing->term;
-                    }
+        // Get a list of which email accounts are associated with this client
+        $module = $this->getModule();
+        $account_emails_meta = $this->ModuleClientMeta->get(
+            isset($vars['client_id']) ? $vars['client_id'] : 0,
+            'cwatch_account_emails',
+            $module->id
+        );
 
-                    // Add licenses to the customer account according to the config options provided
-                    foreach ($license_types as $license_type => $quantity) {
-                        for ($i = 0; $i < $quantity; $i++) {
-                            $license_response = $api->addLicense(
-                                $license_type,
-                                $license_type == 'BASIC_DETECTION' ? 'UNLIMITED' : $license_term,
-                                $vars['cwatch_email'],
-                                $vars['cwatch_firstname'],
-                                $vars['cwatch_lastname'],
-                                $vars['cwatch_country']
-                            );
-
-                            $this->log(
-                                'addlicense',
-                                $license_response->raw(),
-                                'output',
-                                $license_response->status() == 200
-                            );
-
-                            // Break on error
-                            if ($license_response->status() != 200) {
-                                $errors = ['api' => ['internal' => $license_response->errors()]];
-
-                                break 2;
-                            }
-                        }
-                    }
+        $license_keys = [];
+        if (isset($vars['use_module']) && $vars['use_module'] == 'true') {
+            // Add user and licenses
+            $license_keys = $this->pushUserAndLicenses($vars);
+
+            if ($this->Input->errors()) {
+                // Error
+                // Delete the user if this is the only service using this email
+                if (isset($vars['cwatch_email'])
+                    && !$account_emails_meta
+                    || (array_key_exists($vars['cwatch_email'], $account_emails_meta->value)
```

Comment:
You may want to _strtolower_ the email address so you only store and compare email addresses that are lower case. For instance, we wouldn't want to add the same email address that only differs from a capital letter.